### ISSUE TITLE
User upload limits, destroyed subject sets and upload limits.

### ISSUE THREAD

**Issue by camallen** [zooniverse/Panoptes#1037]
The front end allows a user to delete a subject set and the associated subjects.

![image](https://cloud.githubusercontent.com/assets/295329/8404753/73f047b2-1e47-11e5-8b23-c0706e3bc06b.png)

However this doesn't delete the actual subjects, just the subject set and the associated set_member_subjects. Thus their uploaded subject count never decreases and when trying to manage their project data sets, i.e. upload new data, the api checks if the user has < 10K uploaded subjects and error if not. This behaviour will appear to be inconsistent with what they are seeing in the front-end project builder, i.e. possibly 0 subject sets and 0 linked subjects.

I think we need to provide some filtering on the subject table for the uploading user (db index & filter option in the serializer) and link this to a front end interface to manage an uploaders set of subjects so they can see what data is available and delete as they wish. Also we should probably change the wording on the front end when removing a subject set to "Delete this subject set and unlink the 30 subjects".

Pinging @brian-c, @chrissnyder @edpaget for thoughts, also @aliburchard noticed this behaviour with one of her collaborators.

**Comment by twhyntie** [comment]
FWIW I noticed this too, but was in a rush at the time do didn't note it down. I didn't realise subjects weren't actually deleted and the count doesn't get adjusted accordingly.

If user limits are going to be imposed I'd say the ability to delete subjects via the Front End is essential, so :+1: :-)

**Comment by edpaget** [comment]
The problem comes if you upload subjects that are reused in a different project or referenced in Talk. If you delete the subject set all of a sudden all those projects, links, and discussions are broken.

**Comment by camallen** [comment]
I think we should allow deletion of user subjects that have no links to projects to allow the the user to clean up their test data.

**Comment by aliburchard** [comment]
Agreed. As people are beginning the project building stage, they're going to upload subject sets just for testing out the workflow, but that they never need to use when they really push the project live. 

What if I upload someone else grabs a subject that I've uploaded as part of a test set and they use it in their own project or a collection. I want to delete that subject because I am not actually using it and need to open up my 10K limit, but that subject now has links elsewhere. If we (Zooniverse) hold onto that subject somewhere behind the scenes, will that subject still count against my 10K limit even though I've deleted it from *my* project?
